onnxruntime/core/providers/rocm/rocm_allocator.cc

```diff
@@ -65,7 +65,21 @@ void* ROCMExternalAllocator::Alloc(size_t size) {
 }
 
 void ROCMExternalAllocator::Free(void* p) {
+  std::lock_guard<OrtMutex> lock(lock_);
```

Comment:
Inconsistent usage of locks for guarding between ROCM and CUDA - Here the lock does guard both the free and alloc call in Reserve().